internal/stacktrace/stacktrace.go

```diff
@@ -0,0 +1,195 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+//go:generate msgp -o=stacktrace_msgp.go -tests=false
+
+package stacktrace
+
+import (
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/go-secure-stdlib/parseutil"
+)
+
+var enabled = true
+var defaultTopFrameDepth = 8
+var defaultMaxDepth = 32
+
+const defaultCallerSkip = 3
+const stackTraceDepthEnvVar = "DD_APPSEC_MAX_STACK_TRACE_DEPTH"
+const stackTraceDisabledEnvVar = "DD_APPSEC_STACK_TRACE_ENABLE"
+
+func init() {
+	if env := os.Getenv(stackTraceDepthEnvVar); env != "" {
+		if depth, err := parseutil.SafeParseInt(env); err == nil {
+			defaultMaxDepth = depth
+		}
+	}
+
+	defaultTopFrameDepth = defaultMaxDepth / 4
```

Comment:
No big deal, but I wonder if this should ensure it's at minimum set to `1` (if `defaultMaxDepth` is not divisible by 4, this could get to be 0).

Reply:
This behavior of the value being zero does not seem to pose any issue in the code. I would even say that this could end up being bad if `defaultMaxDepth` is one. We would only get the top frame and not the current one (which is more useful)